Common/Data/SubscriptionDataConfigExtensions.cs

```diff
@@ -116,10 +116,18 @@ public static class SubscriptionDataConfigExtensions
         public static bool TickerShouldBeMapped(this SubscriptionDataConfig config)
         {
             // we create an instance of the data type, if it is a custom type
-            // it can override RequiresMapping else it will use security type
+            // it can override RequiresMapping else it will use security type\
+            return config.GetBaseDataInstance().RequiresMapping();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BaseData"/> type defined in <paramref name="config"/> with the symbol properly set
+        /// </summary>
+        public static BaseData GetBaseDataInstance(this SubscriptionDataConfig config)
```

Comment:
As part of this PR, I noticed some cases where we're creating the base data instance but not setting the symbol property. Since methods like `RequiresMapping()` and `IsSparseData()` can depend on symbol information (specifically security type), I went through and updated the easy cases to use this new method and the appropriate `BaseData` method directly. This also had the side effect of removing a few unnecessary `Activator` calls for newing up base data instances we didn't need.